test/e2e/apply_run_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package e2e
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestApplyRun(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	yaml := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: simple-cm
+data:
+  hello_msg: good-morning-bangalore
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: simple-cm1
+data:
+  hello_msg: hello
+`
+	name := "test-apply-run"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+	logger.Section("creating an app with multiple resources", func() {
+		out, _ := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--apply-run"},
+			RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml)})
+		expectedOutput := `
+# add: configmap/simple-cm (v1) namespace: kapp-test
+---
+apiVersion: v1
+data:
+  hello_msg: good-morning-bangalore
+kind: ConfigMap
+metadata:
+  labels:
+    kapp.k14s.io/app: "1654768619446772000"
+    kapp.k14s.io/association: v1.aa6ba70e7f14b3140de8009fda0a6fad
+  name: simple-cm
+  namespace: kapp-test
+# add: configmap/simple-cm1 (v1) namespace: kapp-test
+---
+apiVersion: v1
+data:
+  hello_msg: hello
+kind: ConfigMap
+metadata:
+  labels:
+    kapp.k14s.io/app: "1654768619446772000"
+    kapp.k14s.io/association: v1.df359155db7824da8b7d86ec097a40cf
+  name: simple-cm1
+  namespace: kapp-test
+
+Succeeded
+`
+		out = strings.TrimSpace(replaceTarget(replaceSpaces(replaceTs(out))))
+		// filtering value for label "kapp.k14s.io/app:" and updating with hardcode value
+		if out != "" && len(strings.Split(out, "kapp.k14s.io/app: ")) > 2 {
+			repValue1 := strings.Split((strings.Split(out, "kapp.k14s.io/app: ")[1]), "\n")[0]
```

Comment:
much easier way to replace would be to use a regex instead of splitting strings and assembling them back.